flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -283,15 +284,27 @@ public Builder uidPrefix(String newPrefix) {
         }
       }
 
+      // Find out the equality field id list based on the user-provided equality field column names.
+      List<Integer> equalityFieldIds = Lists.newArrayList();
+      if (equalityFieldColumns != null && equalityFieldColumns.size() > 0) {
+        for (String column : equalityFieldColumns) {
+          org.apache.iceberg.types.Types.NestedField field = table.schema().findField(column);
+          Preconditions.checkNotNull(field, "Missing required equality field column '%s' in table schema %s",
```

Comment:
nit: do we need to include table schema in the error msg?

Reply:
I think this is needed, that is helpful for user to find out the problem.

Reply:
table schema can be very long and make the log difficult to read. user can always inspect table schema using tools like Spark, Trino, Flink etc. You can keep it unless more people think it is not necessary